pilot/pkg/networking/v1alpha3/listener.go

```diff
@@ -45,3 +83,532 @@ func BuildListeners(env model.Environment, node model.Proxy) ([]*xdsapi.Listener
+	// env              model.Environment
+	mesh             *meshconfig.MeshConfig
+	proxy            model.Proxy
+	proxyInstances   []*model.ServiceInstance
+	routeConfig      *xdsapi.RouteConfiguration
+	rdsConfig        *http_conn.HttpConnectionManager_Rds
+	ip               string
+	port             int
+	rds              string
+	useRemoteAddress bool
+	direction        http_conn.HttpConnectionManager_Tracing_OperationName
+	store            model.IstioConfigStore
+	authnPolicy      *authn.Policy
+}
+
+func buildHTTPListener(opts buildHTTPListenerOpts) *xdsapi.Listener {
+	filters := []*http_conn.HttpFilter{}
+	filters = append(filters, &http_conn.HttpFilter{
+		Name: util.CORS,
+	})
+	// TODO: need alphav3 fault filters.
+	// filters = append(filters, buildFaultFilters(opts.config, opts.env, opts.proxy)...)
+	filters = append(filters, &http_conn.HttpFilter{
+		Name: util.Router,
+	})
+
+	/*	TODO(mostrowski): need to port internal build functions for mixer.
+		if opts.mesh.MixerCheckServer != "" || opts.mesh.MixerReportServer != "" {
+			mixerConfig := v1.BuildHTTPMixerFilterConfig(opts.mesh, opts.proxy, opts.proxyInstances, opts.outboundListener, opts.store)
+		filter := &http_conn.HttpFilter{
+			Name: v1.MixerFilter,
+			Config:messageToStruct(mixerConfig),
+		}
+			filters = append([]*http_conn.HttpFilter{filter}, filters...)
+		}
+	*/
+	//refresh := time.Duration(opts.mesh.RdsRefreshDelay.Seconds) * time.Second
+
+	if filter := buildJwtFilter(opts.authnPolicy); filter != nil {
+		filters = append([]*http_conn.HttpFilter{filter}, filters...)
+	}
+
+	var rds *http_conn.HttpConnectionManager_Rds
+	if opts.rds != "" {
+		rds = &http_conn.HttpConnectionManager_Rds{
+			Rds: &http_conn.Rds{
+				RouteConfigName: opts.rds,
+				ConfigSource: core.ConfigSource{
+					ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
+						ApiConfigSource: &core.ApiConfigSource{
+							ApiType:      core.ApiConfigSource_GRPC,
+							ClusterNames: []string{RDSName},
+							//RefreshDelay: &refresh,
+						},
+					},
+				},
+			},
+		}
+	} else {
+		rds = opts.rdsConfig
+	}
+
+	connectionManager := &http_conn.HttpConnectionManager{
+		CodecType: http_conn.AUTO,
+		AccessLog: []*accesslog.AccessLog{
+			{
+				Config: nil,
+			},
+		},
+		HttpFilters:      filters,
+		StatPrefix:       "http",
+		RouteSpecifier:   rds,
+		UseRemoteAddress: &google_protobuf.BoolValue{opts.useRemoteAddress},
+	}
+
+	if opts.mesh.AccessLogFile != "" {
+		fl := &accesslog.FileAccessLog{
+			Path: opts.mesh.AccessLogFile,
+		}
+
+		connectionManager.AccessLog = []*accesslog.AccessLog{
+			{
+				Config: messageToStruct(fl),
+				Name:   fileAccessLog,
+			},
+		}
+	}
+
+	if opts.mesh.EnableTracing {
+		connectionManager.Tracing = &http_conn.HttpConnectionManager_Tracing{
+			OperationName: opts.direction,
+		}
+		connectionManager.GenerateRequestId = &google_protobuf.BoolValue{true}
+	}
+
+	connectionManagerJSON, _ := json.MarshalIndent(connectionManager, "  ", "  ")
```

Comment:
i'm confused about why we are doing this. what is the reason?

Reply:
to print it in the logs.. This code was already there. I just copied it.